src/cart/crt_group.c

```diff
@@ -2569,6 +2569,10 @@ crt_rank_self_set(d_rank_t rank, uint32_t group_version_min)
 	}
 
 	if (default_grp_priv->gp_self != CRT_NO_RANK) {
+		if (default_grp_priv->gp_self == rank) {
+			D_WARN("Self rank was already set to %d, skip\n", rank);
+			D_GOTO(out, rc);
+		}
```

Comment:
Yep, this would also be okay. Thank you very much. If in the future we encounter a situation that the Go code can't deal with, we may reconsider this approach.